c7n/policy.py

```diff
@@ -619,6 +619,157 @@ class EC2InstanceState(LambdaMode):
                      'stopped', 'stopping', 'terminated']}})
 
 
+@execution.register('hub-action')
+@execution.register('hub-finding')
+class SecurityHub(LambdaMode):
+    """
+    Execute a policy lambda in response to security hub finding event or action.
+
+    .. example:
+
+    This policy will provision a lambda and security hub custom action.
+    The action can be invoked on a finding or insight result (collection
+    of findings). The action name will ahve the resource type prefixed.
+
+    .. code-block: yaml
+
+       policy:
+         - name: remediate
+           resource: aws.ec2
+           mode:
+             type: hub-action
+             role: MyRole
+           actions:
+            - snapshot
+            - type: set-instance-profile
+              name: null
+            - stop
+
+    .. example:
+
+    This policy will provision a lambda that will process high alert findings from
+    guard duty (note custodian also has support for guard duty events directly).
+
+    .. code-block: yaml
+
+       policy:
+         - name: remediate
+           resource: aws.iam
+           filters:
+             - type: event
+               key: detail.findings[].ProductFields.aws/securityhub/ProductName
+               value: GuardDuty
+             - type: event
+               key: detail.findings[].ProductFields.aws/securityhub/ProductName
+               value: GuardDuty
+           actions:
+             - remove-keys
+
+    Note, for custodian we support additional resources in the finding via the Other resource,
+    so these modes work for resources that security hub doesn't natively support.
+
+    https://docs.aws.amazon.com/securityhub/latest/userguide/securityhub-cloudwatch-events.html
+    """
+
+    schema = utils.type_schema(
+        'hub-finding', aliases=('hub-action',),
+        # if the mode should be executed for a custom action, this is the list
+        # of action names to filter for.
+        # actions={'type': 'array', 'items': {'type': 'string'}},
+        rinherit=LambdaMode.schema)
+
+    ActionFinding = 'Security Hub Findings - Custom Action'
+    ActionInsight = 'Security Hub Insight Results'
+    ImportFinding = 'Security Hub Findings - Imported'
+
+    handlers = {
+        ActionFinding: 'resolve_action_finding',
+        ActionInsight: 'resolve_action_insight',
+        ImportFinding: 'resolve_import_finding'
+    }
+
+    def validate(self):
+        super(SecurityHub, self).validate()
+        exists = self.policy.resource_manager.action_registry.get('post-finding')
+        if not exists:
+            raise ValueError(
```

Comment:
its unclear if this validation is useful as post-finding is supported for every aws resource.